src/Generators/Microsoft.Gen.Logging/Emission/Emitter.Method.cs

```diff
@@ -276,132 +165,317 @@ static string ConvertPropertyToString(LoggingProperty lp, string arg)
             return $"{arg}{question}.ToString()";
         }
 
-        static string GetException(LoggingMethod lm)
+        static (string exceptionArg, string exceptionLambdaArg) GetException(LoggingMethod lm)
         {
             string exceptionArg = "null";
+            string exceptionLambdaArg = "_";
+
             foreach (var p in lm.Parameters)
             {
                 if (p.IsException)
                 {
                     exceptionArg = p.Name;
+
+                    if (p.UsedAsTemplate)
+                    {
+                        exceptionLambdaArg = lm.GetParameterNameInTemplate(p);
```

Comment:
I thought that we emit a warning when the exception is listed in message template